### PR TITLE
Small changes

### PR DESCRIPTION
**Pull request by fredericopissarra**
Tiny changes for your appreciation...

- mymmap() tests for wrong return value for errors (it returns -1 instead of NULL ptr);
- strerror() (MT-Unsafe) used in web_client_send_deflate;
- Reduce the size of some char buffers and increase others (strerror() and strerror_r() can return an huge string, in other languages than english).

[]s

### CODE REVIEW

File: src/common.c [hunk 1]
```diff
@@ -669,16 +669,17 @@ void *mymmap(const char *filename, size_t size, int flags, int ksm)
...
 					error("Cannot truncate file '%s' to size %ld. Will use the larger file.", filename, size);
 
 #ifdef MADV_MERGEABLE
 				if(flags & MAP_SHARED || !enable_ksm || !ksm) {
 #endif
-					mem = mmap(NULL, size, PROT_READ|PROT_WRITE, flags, fd, 0);
-					if(mem) {
+					/* FIX: mmap() returns MAP_FAILED in case of error, not a NULL pointer! */
+					mmap(NULL, size, PROT_READ|PROT_WRITE, flags, fd, 0);
+					if(mem != MAP_FAILED) {
```
Comment (simonnagl) [inline src/common.c]: `mem` is always null here. You removed the assignment one line above.

**Comment by fredericopissarra** [comment]
@simonnagl, thanks!

**Comment by ktsaou** [comment]
I have adapted your changes to the current source tree, with PR #533 
(I don't know to cherry pick commits yet, so your name is lost - but I added you to the commit message - sorry)

**Comment by fredericopissarra** [comment]
That's all right! :)

**Pull request closed without merging**